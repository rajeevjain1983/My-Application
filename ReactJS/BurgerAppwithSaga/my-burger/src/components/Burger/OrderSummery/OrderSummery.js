import React from 'react';
import Aux from '../../../hoc/Aux/Aux';
import Button from '../../UI/Button/Button';

const orderSummery = (props) =>{
const ingredientsDetails = Object.keys(props.ingredients).map(igkey=>{
    return <li key={igkey}><strong style={{textTransform:'capitalize'}}>{igkey} : </strong>{props.ingredients[igkey]}</li>
})
    return(
        <Aux>
            <h3>Your Order detail!</h3>
            <p>Your delicious burger is ready with following ingredients</p>
            <ul>
                {ingredientsDetails}
            </ul>
            <p><strong>Total Price: {props.totalPrice.toFixed(2)}</strong></p>
            <p>You want to continue</p>
            <Button clicked={props.cancelHander} btnType={'Danger'}>Cancel</Button>
            <Button clicked={props.continueHander} btnType={'Success'}>Continue</Button>

        </Aux>
    )
}
export default orderSummery;