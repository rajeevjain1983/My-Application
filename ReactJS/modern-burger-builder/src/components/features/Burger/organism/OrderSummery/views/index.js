export { default } from "./OrderSummery.view";
