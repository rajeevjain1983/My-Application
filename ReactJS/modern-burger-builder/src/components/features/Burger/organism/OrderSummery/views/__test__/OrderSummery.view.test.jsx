import React from "react";
import Adapter from "enzyme-adapter-react-16";
import { shallow, configure } from "enzyme";
import renderer from "react-test-renderer";

import OrderSummery from "../../../OrderSummery";

configure({ adapter: new Adapter() });

describe("<OrderSummery />", () => {
  const props = {
    ingredients: {
      Meat: 1,
      Bacon: 1,
      Cheese: 1,
    },
    ingredientsHandler: () => {},
    burgerPrice: 23,
    purchasable: true,
    show: true,
    showModal: () => {},
    hideModal: () => {},
  };
  it("Should match snapshot", () => {
    const wrapper = shallow(<OrderSummery {...props} />);

    // const wrapper = renderer.create(<OrderSummery {...props} />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
