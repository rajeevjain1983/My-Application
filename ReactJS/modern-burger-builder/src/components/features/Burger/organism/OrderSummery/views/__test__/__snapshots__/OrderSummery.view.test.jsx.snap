// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderSummery /> Should match snapshot 1`] = `
<OrderSummery
  burgerPrice={23}
  className="withStyle-sc-1y71933-0 kXuyiV"
  hideModal={[Function]}
  ingredients={
    Object {
      "Bacon": 1,
      "Cheese": 1,
      "Meat": 1,
    }
  }
  ingredientsHandler={[Function]}
  purchasable={true}
  show={true}
  showModal={[Function]}
/>
`;
